catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/QueryResponseImpl.java

```diff
@@ -285,7 +290,11 @@ public Result take() {
     private Result handleTake() {
         Result result = null;
         try {
-            result = queue.take();
+            if (timeoutMillis <= 0) {
+                result = queue.take();
```

Comment:
This will allow the current bug to happen if developers send 0 or less..  I think if we  should override to 5 minutes or some other sensible default to avoid a potential indefinite timeout.